app/lib/activitypub/activity/create.rb

```diff
@@ -209,6 +210,41 @@ def process_attachments
     media_attachments
   end
 
+  def process_poll
+    return unless @object['type'] == 'Question' && (@object['anyOf'].is_a?(Array) || @object['oneOf'].is_a?(Array))
+
+    expires_at = begin
+      if @object['closed'].is_a?(String)
+        @object['closed']
+      elsif @object['closed'].is_a?(TrueClass)
+        Time.now.utc
+      else
+        @object['endTime']
+      end
+    end
+
+    if @object['anyOf'].is_a?(Array)
+      multiple = true
+      items    = @object['anyOf']
+    else
+      multiple = false
+      items    = @object['oneOf']
+    end
+
+    Poll.new(
+      account: @account,
+      multiple: multiple,
+      expires_at: expires_at,
+      options: items.map { |item| item['name'].presence || item['content'] },
+      cached_tallies: items.map { |item| item.dig('replies', 'totalItems') || 0 }
+    )
+  end
+
+  def poll_vote?
+    return false if replied_to_status.nil? || replied_to_status.poll.nil? || !replied_to_status.local? || !replied_to_status.poll.options.include?(@object['content'])
+    replied_to_status.poll.votes.create!(account: @account, choice: replied_to_status.poll.options.index(@object['content']))
```

Comment:
Having `poll_vote?` actually process the vote is confusing.

Reply:
Also, we could, at this point, send an `Update` of the polls' replies collection to other instances (with the same rules we use to forward replies, except we don't require LDSignatures) to update the tally.

Reply:
Wasted effort, in my opinion. Here is why: We can't guarantee that an Update will be sent to everyone who ever came across the poll. We do this for normal replies because previously they were missing altogether, and if some branches are not visible, that's not that big of a deal. However, in case of polls, if some votes are missing, that's a gross misrepresentation of the poll and makes it useless/harmful. So, either way, we need to go and download the latest tallies from the origin. Since we need to do that anyway, I see messing with this whole Update business as wasted effort.

Reply:
If we get an `Update` of a poll with updated tallies and a `closed` attribute, we don't have to fetch anything.

Reply:
That being said, such an update mechanism doesn't have to be part of this PR, we can always add it later.